addons/mail/static/src/model/model_field_relation_set.js

```diff
@@ -80,10 +81,12 @@ export class RelationSet {
                                     break;
                                 }
                                 case 'case-insensitive-asc':
-                                    if (valA.toLowerCase() === valB.toLowerCase()) {
+                                    const cleanedValA = cleanSearchTerm(valA);
+                                    const cleanedValB = cleanSearchTerm(valB);
```

Comment:
I believe this will be problematic in terms of lint, switch case must be wrapped in scope to define const inside

Reply:
Scoped !